local/local.go

```diff
@@ -351,8 +351,10 @@ func (i *Image) doSave() (types.ImageInspect, error) {
 			done <- err
 			return
 		}
-		defer res.Body.Close()
-		io.Copy(ioutil.Discard, res.Body)
+		if err := ensureReaderClosed(res.Body); err != nil {
```

Comment:
This solely seems to be discarding the response. How would that work or should this be changed to account for #24?

Reply:
We were already discarding the response on master. The important thing here is to finish reading the entire response and closing before we send on the `done` chan and unblock the next interaction with the docker client (if the reader is not closed the client can't reuse the same connection).

This will have to be thoughtfully merged with #24 to avoid reintroducing the bug.